GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+            // get the current state of the repo
+            await TaskScheduler.Default;
+
+            if (module.IsValidGitWorkingDir())
+            {
+                if (module.InTheMiddleOfBisect())
+                {
+                    await SetStateAsync(State.Bisect, false);
+                }
+                else
+                {
+                    bool conflicted = module.InTheMiddleOfConflictedMerge();
+
+                    if (module.InTheMiddleOfRebase())
+                    {
+                        await SetStateAsync(State.Rebase, conflicted);
+                    }
+                    else if (module.InTheMiddleOfMerge())
+                    {
+                        await SetStateAsync(State.Merge, conflicted);
+                    }
+                    else if (module.InTheMiddleOfPatch())
+                    {
+                        await SetStateAsync(State.Patch, conflicted);
+                    }
+                    else
+                    {
+                        await SetStateAsync(State.None, false);
+                    }
+                }
+            }
+
+            async Task SetStateAsync(State state, bool conflicted)
+            {
+                if ((state == _state) && (conflicted == _hasConflicts))
+                {
+                    // nothing to do
+                    return;
+                }
+
+                _state = state;
+                _hasConflicts = conflicted;
+
+                // now update in main UI thread
+                await this.SwitchToMainThreadAsync();
+
+                // remove old controls
+                ButtonContainer.Controls.Clear();
+
+                string action = "";
+
+                switch (_state)
+                {
+                    case State.Bisect:
+                        action = "Bisect";
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Rebase:
+                        action = "Rebase";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Merge:
+                        action = "Merge";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        break;
+                    case State.Patch:
+                        action = "Patch";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                }
+
+                if (_state != State.None)
```

Comment:
Please invert and return early.

Also the code after line:77 isn't changing the value of `_state`. So why are we not calling `Hide()` and exiting after line:77?